### PR TITLE
prevent early stream termination in async http request

### PR DESCRIPTION
**Pull request by iidebyo**
### What does this PR do?

1. add http request async iterator test
2. remove unnecessary destroy() in http incoming message

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

- [x] I included a test for the new code, or an existing test covers it

### CODE REVIEW

**Comment by iidebyo** [comment]
see #5081

File: test/js/node/http/node-http.test.ts [hunk 1]
```diff
@@ -219,6 +219,36 @@ describe("node:http", () => {
       });
     }
 
+    it("should be async readable", async done => {
```
Comment (Jarred-Sumner) [inline test/js/node/http/node-http.test.ts:222]: this test doesn't work 

the builtins were never re-generated, which means the changes to the JS were not reflected, but this test did not fail

Comment (iidebyo) [reply test/js/node/http/node-http.test.ts:222]: It’s intermittent. I’m not sure what the correct size should be atm.

Edit: 40k is consistent locally with wsl2.

Comment (iidebyo) [reply test/js/node/http/node-http.test.ts:222]: n > 16384 or no error occurs.

Comment (iidebyo) [reply test/js/node/http/node-http.test.ts:222]: was it hanging on the test server? i think all it needed was a reject.

**Comment by asilvas** [comment]
any update on this?

**Comment by iidebyo** [comment]
working for me w/ 1.0.21+837cbd60d

**Comment by Jarred-Sumner** [comment]
This was done sometime later

**Pull request closed without merging**